tests/clock/ClockUtils.py

```diff
@@ -0,0 +1,384 @@
+                pytest_assert(False, f'Given string "{time_str}" is not a valid time')
+
+    @staticmethod
+    def get_valid_timezones(duthosts):
+        """
+        @summary:
+            Get the list of valid timezones from 'show clock timezones' command
+        @param duthosts: duthosts object
+        @return: list of timezones (strings)
+        """
+        with allure.step('Get list of valid timezones from show clock timezones command'):
+            logging.info('Get list of valid timezones from show clock timezones command')
+            return ClockUtils.run_cmd(duthosts=duthosts, cmd=ClockConsts.CMD_SHOW_CLOCK_TIMEZONES).split()
+
+    @staticmethod
+    def validate_timezone(timezone_str, duthosts):
+        """
+        @summary:
+            Verify that the given string is an abbreviation of a valid timezone
+        @param timezone_str: the given string
+        @param duthosts: duthosts
+        """
+        with allure.step(f'Verify that given string "{timezone_str}" is a valid timezone'):
+            logging.info(f'Verify that given string "{timezone_str}" is a valid timezone')
+
+            with allure.step('Get timezone from timedatectl linux command'):
+                logging.info('Get timezone from timedatectl linux command')
+                timedatectl_output = \
+                    ClockUtils.parse_linux_cmd_output(ClockUtils.run_cmd(duthosts, ClockConsts.CMD_TIMEDATECTL))
+                timedatectl_timezone = timedatectl_output[ClockConsts.TIME_ZONE]
+                timedatectl_tz_name = timedatectl_timezone.split()[0]
+                timedatectl_tz_abbreviation = timedatectl_timezone.split(' ')[1].split(',')[0].replace('(', '')
+                logging.info(f'Timezone in timedatectl: "{timedatectl_timezone}"\nTimezone name: '
+                             f'"{timedatectl_tz_name}"\nTimezone abbreviation: "{timedatectl_tz_abbreviation}" ')
+
+            with allure.step(f'Verify timezone name "{timedatectl_tz_name}" from timedatectl is in valid timezones'):
+                logging.info(f'Verify timezone name "{timedatectl_tz_name}" from timedatectl is in valid timezones')
+                valid_timezones = ClockUtils.get_valid_timezones(duthosts)
+                pytest_assert(timedatectl_tz_name in valid_timezones,
+                              f'Error: string "{timezone_str}" is not in the valid timezones list')
+
+            with allure.step(f'Verify that the given timezone "{timezone_str}" equals to timezone abbreviation in '
+                             f'timedatectl "{timedatectl_tz_abbreviation}"'):
+                logging.info(f'Verify that the given timezone "{timezone_str}" equals to timezone abbreviation in '
+                             f'timedatectl "{timedatectl_tz_abbreviation}"')
+                ClockUtils.verify_value(expected=timedatectl_tz_abbreviation, actual=timezone_str)
+
+    @staticmethod
+    def verify_value(expected, actual, should_be_equal=True):
```

Comment:
I don't think we need a specific function to check this. This can be validated directly in the test flow.

Reply:
you are correct basically, but i wanted in all of these value validations to have this format of error and log messages. so it was much cleaner to create this function, and use it in the different places. 
did it convince you?

Reply:
I feel this has introduced more redundant lines and steps which makes the test more verbose than it should be. This will affect the readability of the code. However I would request feedback from @roy-sror whether it makes sense to have separate APIs to do validations for the purpose of logging.

Reply:
i removed these functions. will be noticed in the following commit tomorrow